Sources/SwiftFormatPrettyPrint/TokenStreamCreator.swift

```diff
@@ -1256,7 +1256,16 @@ fileprivate final class TokenStreamCreator: SyntaxVisitor {
   }
 
   override func visit(_ node: ReturnClauseSyntax) -> SyntaxVisitorContinueKind {
-    after(node.arrow, tokens: .space)
+    if node.parent?.is(FunctionTypeSyntax.self) ?? false {
+      // `FunctionTypeSyntax` used to not use `ReturnClauseSyntax` and had 
+      // slightly different formatting behavior than the normal 
+      // `ReturnClauseSyntax`. To maintain the previous formatting behavior, 
+      // add a special case.
```

Comment:
@allevato I leave it up to you to decide whether you want to get rid of this special case in a follow-up PR.